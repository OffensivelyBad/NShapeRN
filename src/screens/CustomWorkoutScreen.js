import React, { PureComponent } from 'react';
import { View, FlatList } from 'react-native';
import { SimpleCell } from '../components/common/';

class CustomWorkoutScreen extends PureComponent {

  static navigationOptions = {
    title: 'Custom Workout',
  }

  constructor(props) {
    super(props);
    this.state = {
      cells: [
        {id: '1', title: 'Workout 1', screen: 'WorkoutPreview', hasSwitch: false, toggled: false },
        {id: '2', title: 'Workout 2', screen: 'WorkoutPreview', hasSwitch: false, toggled: false },
        {id: '3', title: 'Workout 3', screen: 'WorkoutPreview', hasSwitch: false, toggled: false },
      ]
    }
  }

  _navigateToScreen = (route) => {
    const navigateAction = this.props.navigation.navigate({
      routeName: route
    });
    this.props.navigation.dispatch(navigateAction);
  }

  _onItemTap = (item) => {
    if (item.screen) {
      this._navigateToScreen(item.screen);
    }
  }

  _keyExtractor = (item) => item.id;

  _renderItem = ({item}) => {
    return (
      <SimpleCell
        key={item.id}
        item={item}
        onPressItem={this._onItemTap}
        backgroundColor='#fff'
        textColor='#333'
        textMarginLeft={30}
        height={50}
        fontSize={30}
        hasSwitch={item.hasSwitch}
      >
        {item.title}
      </SimpleCell>
    );
  }

  render() {
    return (
      <View style={styles.containerStyle}>
        <FlatList
          keyExtractor={this._keyExtractor}
          data={this.state.cells}
          renderItem={this._renderItem}
        />
      </View>
    );
  }

}

const styles = {
  containerStyle: {
    flex: 1
  }
}

export default CustomWorkoutScreen;
