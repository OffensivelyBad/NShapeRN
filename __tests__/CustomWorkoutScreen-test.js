import React from 'react';
import CustomWorkoutScreen from '../src/screens/CustomWorkoutScreen';
import renderer from 'react-test-renderer';

test('CustomWorkoutScreen renders correctly', () => {
    const tree = renderer.create(<CustomWorkoutScreen />).toJSON();
    expect(tree).toMatchSnapshot();
});
