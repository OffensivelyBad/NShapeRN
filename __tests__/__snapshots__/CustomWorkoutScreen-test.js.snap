// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomWorkoutScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "hasSwitch": false,
            "id": "1",
            "screen": "WorkoutPreview",
            "title": "Workout 1",
            "toggled": false,
          },
          Object {
            "hasSwitch": false,
            "id": "2",
            "screen": "WorkoutPreview",
            "title": "Workout 2",
            "toggled": false,
          },
          Object {
            "hasSwitch": false,
            "id": "3",
            "screen": "WorkoutPreview",
            "title": "Workout 3",
            "toggled": false,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": 1,
                "borderColor": "#ddd",
                "flex": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "flex": 2,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#333",
                  },
                  Object {
                    "fontSize": 30,
                  },
                  Object {
                    "alignSelf": undefined,
                  },
                  Object {
                    "marginLeft": 50,
                  },
                ]
              }
            >
              Workout 1
            </Text>
            <Text
              style={
                Object {
                  "flex": 1,
                  "fontWeight": "300",
                }
              }
            />
            <Icon
              color="#ddd"
              name="angle-right"
              size={30}
              style={
                Object {
                  "marginRight": 10,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": 1,
                "borderColor": "#ddd",
                "flex": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "flex": 2,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#333",
                  },
                  Object {
                    "fontSize": 30,
                  },
                  Object {
                    "alignSelf": undefined,
                  },
                  Object {
                    "marginLeft": 50,
                  },
                ]
              }
            >
              Workout 2
            </Text>
            <Text
              style={
                Object {
                  "flex": 1,
                  "fontWeight": "300",
                }
              }
            />
            <Icon
              color="#ddd"
              name="angle-right"
              size={30}
              style={
                Object {
                  "marginRight": 10,
                }
              }
            />
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": 1,
                "borderColor": "#ddd",
                "flex": 1,
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "space-between",
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "flex": 2,
                    "fontWeight": "400",
                  },
                  Object {
                    "color": "#333",
                  },
                  Object {
                    "fontSize": 30,
                  },
                  Object {
                    "alignSelf": undefined,
                  },
                  Object {
                    "marginLeft": 50,
                  },
                ]
              }
            >
              Workout 3
            </Text>
            <Text
              style={
                Object {
                  "flex": 1,
                  "fontWeight": "300",
                }
              }
            />
            <Icon
              color="#ddd"
              name="angle-right"
              size={30}
              style={
                Object {
                  "marginRight": 10,
                }
              }
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
